### PR TITLE
Documentation Cleanup - AWS

### PR DESCRIPTION
**Pull request by Piszmog**
Part of [#42471](https://github.com/sourcegraph/sourcegraph/issues/42471).

This PR ensures all variables and outputs have documentation (`description`). A section was also added to the `README` to include a link to the setup docs.

### Test plan

No tests. This just updates documentation.

### CODE REVIEW

**Review by BolajiOlajide — APPROVED** [review]

File: modules/networking/outputs.tf [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 output "subnet_id" {
-  value = var.nat == true ? aws_subnet.private.0.id : aws_subnet.default.id
+  value       = var.nat == true ? aws_subnet.private.0.id : aws_subnet.default.id
+  description = "The subnet to run the VM in."
 }
 
 output "ip_cidr" {
-  value = local.ip_cidr
+  value       = local.ip_cidr
+  description = "The internal address that is owned by the subnetwork."
```
Comment (eseliger) [inline modules/networking/outputs.tf:13]: I don't fully understand that description, maybe something along these lines?

```suggestion
  description = "The CIDR range of the subnet in which the executor machines will be spawned."
```

File: modules/networking/outputs.tf [hunk 2]
```diff
@@ -1,15 +1,19 @@
 output "vpc_id" {
-  value = aws_vpc.default.id
+  value       = aws_vpc.default.id
+  description = "The ID of the VPC that hosts the cache and the executors in."
 }
 
 output "subnet_id" {
-  value = var.nat == true ? aws_subnet.private.0.id : aws_subnet.default.id
+  value       = var.nat == true ? aws_subnet.private.0.id : aws_subnet.default.id
+  description = "The subnet to run the VM in."
```
Comment (eseliger) [inline modules/networking/outputs.tf:8]:
```suggestion
  description = "The subnet to run the VMs in."
```

**Review by eseliger — APPROVED** [review]
Nice :)

File: modules/networking/outputs.tf (hunk 1 above)
Comment (Piszmog) [reply modules/networking/outputs.tf:13]: I actually just took the docs from `ip_cidr_range` on [google_compute_subnetwork](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_subnetwork)

Comment (eseliger) [reply modules/networking/outputs.tf]: lol, good docs :D do you unterstand what they mean? maybe it's just me then.

Comment (Piszmog) [reply modules/networking/outputs.tf]: I had to search online a bit to understand

**Pull request merged**